parser/tests/integration/test_parser_integration.py

```diff
@@ -0,0 +1,95 @@
+import pytest
+import pathlib
+from openfoodfacts_taxonomy_parser import parser
+
+# taxonomy in text format : test.txt
+TEST_TAXONOMY_TXT = str(pathlib.Path(__file__).parent.parent / "data" / "test")
+
+
+@pytest.fixture
+def new_session():
+    x = parser.Parser()
+    # delete all the nodes and relations in the database
+    query="MATCH (n) DETACH DELETE n"
+    x.session.run(query)
+    return x
+
+
+def test_calling(new_session):
+    x=new_session
+
+    #Create node test
+    x.create_nodes(TEST_TAXONOMY_TXT)
+
+    # total number of nodes
+    query="MATCH (n) RETURN COUNT(*)"
+    result = x.session.run(query)
+    number_of_nodes = result.value()[0]
+    assert number_of_nodes == 13
+
+    # header correctly added
+    query="MATCH (n) WHERE n.id = '__header__' RETURN n.preceding_lines"
+    result = x.session.run(query)
+    header = result.value()[0]
+    assert header == ['# test taxonomy']
+
+    # comment / preceding lines correctly added
+    query="MATCH (n:ENTRY) WHERE size(n.preceding_lines)>0 RETURN n.id,n.preceding_lines"
+    result = x.session.run(query)
+    nodes = result.values()
+    number_of_nodes = len(nodes)
+    assert number_of_nodes == 1
+    assert nodes[0][0] == 'en:meat'
+    assert nodes[0][1] == ['# meat','']
```

Comment:
You should test at least three entry nodes and assert more than their id.

Also check one synonym and one stopwords.